libs/upd/components/src/lib/charts/charts.component.ts

```diff
@@ -153,8 +157,12 @@ export class ChartsComponent implements OnInit {
       this.applyDimensions();
     }
 
-    this.currentLang$.subscribe((nextLang) => {
-      this.lang = nextLang;
+    this.i18n.service.onLangChange(({ lang }) => {
+      this.lang = lang as LocaleId;
+    });
+
+    this.currentLang$.subscribe((lang) => {
+      this.lang = lang === EN_CA ? 'en' : 'fr';
```

Comment:
These are both going to change at the same time, but not always in the same order and will assign the same variable different values. Whatever you're trying to do here, it won't work properly. Just use a single subscription. And try not to use onLangChange either, because that doesn't go through the NgRx store and that's what we want to use as the single source of truth.

Reply:
Thanks for catching that, removed the onLangChange :)